bindings/go/plugin/client/sdk/plugin.go

```diff
@@ -291,3 +302,66 @@ func (p *Plugin) Shutdown(w http.ResponseWriter, _ *http.Request) {
 		p.logger.ErrorContext(p.baseCtx, "Error shutting down plugin", "error", err)
 	}
 }
+
+// performCleanUp looks for a lock file that contains the pid of the process using the corresponding socket file.
+// if the process is not found, we remove the socket file AND the lock file.
+func (p *Plugin) performCleanUp(loc string) error {
+	lockFile := loc + ".lock"
+	if _, err := os.Stat(lockFile); err != nil {
+		return fmt.Errorf("lock file does not exists unable to determine using process: %w", err)
+	}
+
+	content, err := os.ReadFile(lockFile)
+	if err != nil {
+		return fmt.Errorf("could not read lock file: %w", err)
+	}
+
+	pid, err := strconv.Atoi(string(content))
+	if err != nil {
+		return fmt.Errorf("could not parse PID: %w", err)
+	}
+
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return p.removeFiles(loc, lockFile)
+	}
+
+	// Check if the process is actually running by sending signal 0
+	if err := process.Signal(syscall.Signal(0)); err != nil {
```

Comment:
This looks off! is this really correct on any os? if so LGTM

Reply:
Fixed. thanks.